clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -280,14 +274,17 @@ public KafkaStoreIngestionService(
 
     KafkaClusterBasedRecordThrottler kafkaClusterBasedRecordThrottler =
         new KafkaClusterBasedRecordThrottler(kafkaUrlToRecordsThrottler);
+    VeniceProperties clusterProperties = veniceConfigLoader.getVeniceClusterConfig().getClusterProperties();
 
     TopicManagerContext topicManagerContext =
         new TopicManagerContext.Builder().setPubSubTopicRepository(pubSubTopicRepository)
             .setMetricsRepository(metricsRepository)
             .setTopicOffsetCheckIntervalMs(serverConfig.getTopicOffsetCheckIntervalMs())
             .setPubSubPropertiesSupplier(this::getPubSubSSLPropertiesFromServerConfig)
-            .setPubSubAdminAdapterFactory(pubSubClientsFactory.getAdminAdapterFactory())
-            .setPubSubConsumerAdapterFactory(pubSubClientsFactory.getConsumerAdapterFactory())
+            .setPubSubAdminAdapterFactory(
+                PubSubClientsFactory.fromVeniceProperties(clusterProperties).getAdminAdapterFactory())
```

Comment:
same: Let’s avoid creating factories at individual call sites. Instead, we should create the factory
once per server component and reuse it.